fetch_data.py

```diff
@@ -283,3 +301,141 @@ def get_from_dict(data_dict, map_list):
+    """
+    is_return_list = True if result_list is None else False
+    if is_return_list:
+        result_list = []
+    is_create_new = True if in_line_list is None else False
+    for k, v in dict_.items():
+        if is_create_new:
+            in_line_list = []
+        out_line_list = copy.deepcopy(in_line_list)
+        out_line_list.append(k)
+        if not isinstance(v, dict) or len(v) == 0:
+            result_list.append(out_line_list)
+        else:
+            extract_nested_dict_keys(dict_[k], result_list, out_line_list)
+    if is_return_list:
+        return result_list
+
+
+def create_csv(packaged_data, fieldnames, package_coords, fill_di=None):
+    """
+    Returns a CSV version of the fetched data, as a string.
+
+    Args:
+        packaged_data (json): JSON-like data pakage output
+            from the run_fetch_* and run_aggregate_* functions
+        fieldnames (list): list of string values for CSV column names
+            in desired column order
+        package_coord (list): list of string values corresponding to 
+            levels of the packaged_data dict. Should be a subset of fieldnames arg.
+        fill_di (dict): dict to fill in columns with fixed values. 
+            Keys should specify the field name and value should be the 
+            value to fill
+        varname (str): str value used for variable name. Appended 
+        stat (str): str value used for stat name. Appended
+            after fields and before value.
+
+    Returns:
+        string of CSV data
+    """
+    output = io.StringIO()
+    writer = csv.DictWriter(output, fieldnames=fieldnames)
+    writer.writeheader()
+    # extract the coordinate values stored in keys. assumes uniform structure
+    # across entire data package (i.e. n levels deep where n == len(fieldnames))
+    data_package_coord_combos = extract_nested_dict_keys(packaged_data)
+    for coords in data_package_coord_combos:
+        row_di = {}
+        # need more general way of handling fields to be inserted before or after
+        # what are actually available in packaged dicts
+        for field, coord in zip(package_coords, coords):
+            row_di[field] = coord
+        # fill in columns with fixed values if specified
+        if fill_di:
+            for fieldname, value in fill_di.items():
+                row_di[fieldname] = value
+        # write the actual value
+        row_di["value"] = get_from_dict(packaged_data, coords)
+        writer.writerow(row_di)
+    return output.getvalue()
+
+
+def return_csv(csv_data):
+    """Return the CSV data as a download
+
+    Args:
+        csv_data (?): csv data created with create_csv() function
+
+    Returns:
+        CSV Response
+    """
+    response = Response(
+        csv_data,
+        mimetype="text/csv",
+        headers={
+            "Content-Type": 'text/csv; name="climate.csv"',
+            "Content-Disposition": 'attachment; filename="climate.csv"',
```

Comment:
I don't know that it is necessary now but we  may want an optional argument for the `filename` and `name` key-value pairs in the `headers` dictionary.